Tests/Client/RequestRecorderURLProtocol.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  MockURLProtocol.swift
+//  StreamChatClientTests
+//
+//  Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+import XCTest
+
+/// This URLProtocol subclass allows to intercept the network communication
+/// and provides the latest network request made.
+class RequestRecorderURLProtocol: URLProtocol {
+
+    /// Returns the latest network request this URLProtocol recorded.
+    ///
+    /// If no request has been made since the last time this function was invoked, it synchronously
+    /// waits for the next request to be made.
+    ///
+    /// - Parameter timeout: Specifies the time the function waits for a new request
+    /// to be made.
+    static func getLatestRequestSynchronously(timeout: TimeInterval) -> URLRequest? {
+        defer {
+            // Delete the used request
+            latestRequest = nil
+        }
+
+        guard latestRequest == nil else { return latestRequest }
+
+        latestRequestExpectation = .init(description: "Wait for incoming request.")
+        _ = XCTWaiter.wait(for: [latestRequestExpectation!], timeout: 1)
+        return latestRequest
+    }
+
+    override class func canInit(with request: URLRequest) -> Bool {
+        DispatchQueue.main.async {
+            latestRequest = request
+            latestRequestExpectation?.fulfill()
+
+        }
+        return false
+    }
+
+    override class func canonicalRequest(for request: URLRequest) -> URLRequest {
+        // Overriding this function is required by the superclass.
+        return request
+    }
+
+    private static var latestRequestExpectation: XCTestExpectation?
+    private static var latestRequest: URLRequest?
```

Comment:
I see that your style is putting variables after functions, in itself its consistent:
- static func
- overridden static func
- static variables
- instance func
- overridden instance func
- instance vars

putting vars at the end of file makes reading harder IMHO (because you could use the vars inside the functions) and it's also the opposite of what our SDK uses (vars at the top)

Reply:
I agree with @b-onc 
Please put vars on top and functions after. Static stuff first, instance after.

Reply:
fixed